vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -263,24 +263,61 @@ BaseType_t WIFI_IsConnected( void )
 
 WIFIReturnCode_t WIFI_Off( void )
 {
-    return eWiFiSuccess;
+    esp_err_t ret;
+    if( xSemaphoreTake( xWiFiSem, xSemaphoreWaitTicks ) == pdTRUE )
+    {
+        if (wifi_conn_state == true) {
+            ret = esp_wifi_disconnect();
+            if (ret == ESP_OK) {
+                // Wait for wifi disconnected event
+                xEventGroupWaitBits(wifi_event_group, DISCONNECTED_BIT, pdTRUE, pdFALSE, portMAX_DELAY);
+            } else {
+                ESP_LOGE(TAG, "%s:Failed to disconnect wifi %d", __func__, ret);
+                goto err;
+            }
+        }
+
+        if ((ret = esp_wifi_deinit()) != ESP_OK) {
+            if (ret == ESP_ERR_WIFI_NOT_STOPPED) {
+                ret = esp_wifi_stop();
+                if (ret != ESP_OK) {
+                    ESP_LOGE(TAG, "%s:Failed to stop wifi %d", __func__, ret);
+                    goto err;
+                }
+                if (esp_wifi_deinit() != ESP_OK) {
+                    ESP_LOGE(TAG, "%s:Failed to deinit %d", __func__, ret);
+                    goto err;
+                }
+                vEventGroupDelete(wifi_event_group);
```

Comment:
`wifi_event_group` is not deleted if first call to `esp_wifi_deinit()` returns `ESP_OK`

Reply:
Ohh, right. Changed accordingly!